nav2_lifecycle_manager/src/lifecycle_manager.cpp

```diff
@@ -79,18 +76,27 @@ LifecycleManager::LifecycleManager()
   transition_label_map_[Transition::TRANSITION_UNCONFIGURED_SHUTDOWN] =
     std::string("Shutting down ");
 
-  createLifecycleServiceClients();
-
-  if (autostart_) {
-    startup();
-  }
+  init_timer_ = this->create_wall_timer(
+    std::chrono::milliseconds(100),
+    std::bind(&LifecycleManager::init, this));
 }
 
 LifecycleManager::~LifecycleManager()
 {
   RCLCPP_INFO(get_logger(), "Destroying %s", get_name());
 }
 
+void
+LifecycleManager::init()
+{
+  init_timer_->cancel();
```

Comment:
This should be a lambda instead since its very small, here's an example https://github.com/ros2/examples/blob/master/rclcpp/topics/minimal_publisher/lambda.cpp#L35